packages/@aws-cdk/aws-route53-targets/lib/global-accelerator-target.ts

```diff
@@ -0,0 +1,36 @@
+import * as globalaccelerator from '@aws-cdk/aws-globalaccelerator';
+import * as route53 from '@aws-cdk/aws-route53';
+
+/**
+ * Use a Global Accelerator domain name as an alias record target.
+ */
+export class GlobalAcceleratorTarget implements route53.IAliasRecordTarget {
+  /**
+   * The hosted zone Id if using an alias record in Route53.
+   * This value never changes.
+   * Ref: https://docs.aws.amazon.com/general/latest/gr/global_accelerator.html
+   */
+  public static readonly GLOBAL_ACCELERATOR_ZONE_ID = 'Z2BJ6XQ5FK7U4H';
+
+  /**
+   * Create an Alias Target for a Global Accelerator.
+   *
+   * If passing a string value, it must be a valid domain name for an existing Global Accelerator. e.g. xyz.awsglobalaccelerator.com
+   * If passing an instance of an accelerator created within CDK, the accelerator.dnsName property will be used as the target.
+   */
+  constructor(private readonly accelerator: string | globalaccelerator.IAccelerator) {
```

Comment:
Unfortunately, we can't use type unions in our public APIs as they cannot be strongly-modeled in all languages the CDK supports.

Solutions:
1. Typically we [recommend](https://github.com/aws/aws-cdk/blob/master/DESIGN_GUIDELINES.md#unions) using static methods instead (e.g., `GlobalAcceleratorTarget.domain('xyz.awsglobalaccelerator.com')`, `GlobalAcceleratorTarget.accelerator(accelerator)`).
2. Within this package, it looks like maybe just creating a second class like `ApiGateway`/`ApiGatewayDomain` is an existing, alternative pattern. So you could create `GlobalAcceleratorTarget` and `GlobalAcceleratorDomainTarget`. 
3. Final alternative would be to just create this `GlobalAcceleratorTarget` that accepts an `IAccelerator`, and add the `GlobalAcceleratorDomainTarget` (that takes the domain name/string) later based on feedback from the community.

I think given the existing patterns in this module options 2 or 3 are the principal of least surprise; I'm happy either way.

Reply:
Went with option 2. I think it actually makes the code a little cleaner, although it might not be as intuitive to locate and use.